homeassistant/components/osoenergy/water_heater.py

```diff
@@ -0,0 +1,237 @@
+    ATTR_V40MIN,
+    DOMAIN,
+    EXTRA_HEATER_ATTR,
+    HEATER_MAX_TEMP,
+    HEATER_MIN_TEMP,
+    MANUFACTURER,
+    OPERATION_LIST,
+    OSO_ENERGY_TO_HASS_STATE,
+    SERVICE_SET_PROFILE,
+    SERVICE_SET_V40MIN,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up OSO Energy heater based on a config entry."""
+    osoenergy = hass.data[DOMAIN][entry.entry_id]
+    devices = osoenergy.session.device_list.get("water_heater")
+    entities = []
+    if devices:
+        for dev in devices:
+            entities.append(OSOEnergyWaterHeater(osoenergy, dev))
+    async_add_entities(entities, True)
+
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        SERVICE_TURN_ON,
+        {vol.Required(ATTR_UNTIL_TEMP_LIMIT): vol.All(cv.boolean)},
+        "async_oso_turn_on",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_TURN_OFF,
+        {vol.Required(ATTR_UNTIL_TEMP_LIMIT): vol.All(cv.boolean)},
+        "async_oso_turn_off",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_V40MIN,
+        {
+            vol.Required(ATTR_V40MIN): vol.All(
+                vol.Coerce(float), vol.Range(min=200, max=550)
+            ),
+        },
+        "async_set_v40_min",
+    )
+
+    service_set_profile_schema = {}
+    for hour in range(24):
+        service_set_profile_schema[vol.Optional(f"hour_{hour:02d}")] = vol.All(
+            vol.Coerce(int), vol.Range(min=10, max=75)
+        )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_PROFILE,
+        service_set_profile_schema,
+        "async_set_profile",
+    )
+
+
+def _get_utc_hour(local_hour: int):
+    """Get the utc hour."""
+    now = dt_util.now()
+    local_time = now.replace(hour=local_hour, minute=0, second=0, microsecond=0)
+    utc_hour = dt_util.as_utc(local_time)
+    return utc_hour.hour
+
+
+def _get_local_hour(utc_hour: int):
+    """Get the local hour."""
+    now = dt_util.utcnow()
+    now_local = dt_util.now()
+    utc_time = now.replace(hour=utc_hour, minute=0, second=0, microsecond=0)
+    local_hour = dt_util.as_local(utc_time)
+    local_hour = local_hour.replace(day=now_local.day)
+    return local_hour
+
+
+def _convert_profile_to_local(values):
+    """Convert UTC profile to local."""
+    profile = [None] * 24
+    for hour in range(24):
+        local_hour = _get_local_hour(hour)
+        local_hour_string = local_hour.strftime("%Y-%m-%dT%H:%M:%S%z")
+        profile[local_hour.hour] = {local_hour_string: values[hour]}
+
+    return profile
+
+
+class OSOEnergyWaterHeater(OSOEnergyEntity, WaterHeaterEntity):
+    """OSO Energy Water Heater Device."""
+
+    _attr_name = None
+    _attr_operation_list = OPERATION_LIST
+    _attr_supported_features = WaterHeaterEntityFeature.TARGET_TEMPERATURE
```

Comment:
Why do you set an operation list when your water heater doesn't support the operation_mode feature?

Reply:
You are right. I have removed the operation list